### PR TITLE
feat: websockets support

### PR DESCRIPTION
**Pull request by richard-ramos**
Adds the `websockets_param` attribute to the node config to enable ws and wss support

### CODE REVIEW

File: waku-bindings/src/node/config.rs [hunk 1]
```diff
@@ -195,6 +201,26 @@ pub struct GossipSubParams {
...
+    /// Listening address for websocket connections. Default `0.0.0.0`
+    #[default(Some(std::net::IpAddr::V4(std::net::Ipv4Addr::new(0, 0, 0, 0))))]
+    pub host: Option<std::net::IpAddr>,
+    /// TCP listening port for websocket connection. Use `0` for **random**. Default `60001`, if secure websockets support is enabled, the default is `6443“`
+    pub port: Option<usize>,
+    /// Enable secure websockets support
+    #[default(Some(false))]
+    pub secure: Option<bool>,
+    /// Secure websocket certificate path
+    pub cert_path: Option<String>,
```
Comment (chaitanyaprem) [inline waku-bindings/src/node/config.rs:219]: Better to indicate that cert and key are mandatory if secure is set to true.

**Review by chaitanyaprem — APPROVED** [review]
Minor comment.

LGTM

File: waku-bindings/src/node/filter.rs [hunk 2]
```diff
@@ -28,7 +28,7 @@ pub fn waku_filter_subscribe(
         .into_raw();
 
     let result_ptr = unsafe {
-        let result_ptr = waku_sys::waku_filter_subscribe(
+        let result_ptr = waku_sys::waku_legacy_filter_subscribe(
```
Comment (danielSanchezQ) [inline waku-bindings/src/node/filter.rs:31]: Why is this legacy now? And what is the newest call?

Comment (richard-ramos) [reply waku-bindings/src/node/filter.rs:31]: The newest calls can be seen here: https://github.com/waku-org/go-waku/blob/master/library/README.md#waku-filter . These run on a protocol different from filterv1 and are not compatible. Eventually these filter functions should be deprecated and new filter functions using filterv2 must be added.

Comment (richard-ramos) [reply waku-bindings/src/node/filter.rs:31]: Specs for filterV2 can be read here: https://rfc.vac.dev/spec/12/

**Review by danielSanchezQ — APPROVED** [review]

**Pull request merged**